tools/GenerateNewControlInProject.ps1

```diff
@@ -0,0 +1,84 @@
+[CmdLetBinding()]
+Param(
+    [Parameter(Mandatory = $true)]
+    [string]$controlName,
+    [string]$projectName
+)
+
+function FindAndReplaceInFile
+{
+    Param($file, $regex, $replace)
+
+    $content = Get-Content $file -Raw
+
+    [regex]$pattern = $regex
+    $content = $pattern.replace($content, $replace, 1)
+
+    Set-Content -Path $file -Value $content
+}
+
+function AddAttribute
+{
+    Param([xml]$xml, $element, $attrName, $attrValue)
+
+    $attribute = $xml.CreateAttribute($attrName)
+    $attribute.Value = $attrValue;
+    $element.Attributes.Append($attribute)
+}
+
+$toolsDir = Split-Path -Path $MyInvocation.MyCommand.Path;
+$muxControlsDir = Split-Path $toolsDir -Parent
+$controlDir = $muxControlsDir + "\dev\$projectName"
+
+# TODO: check if project directory exists
+
```

Comment:
Test-Path ? or is this TODO more about "what should we do if it exists already?"